src/main.rs

```diff
@@ -69,21 +75,76 @@ impl Group {
 struct DWebBackend {
     path: String,
     port: u16,
+    store: Store,
+    veilid_api: Option<VeilidAPI>,
 }
 
 impl DWebBackend {
-    pub fn new(path: String, port: u16) -> Self {
-        DWebBackend { path, port }
-    }
-    pub async fn start(&mut self) -> Result<()> {
-        // Init veilid
-        println!("Starting in {} with port {}", self.path, self.port);
+    pub fn new(base_path: &str, port: u16) -> Self {
+        let store_path = format!("{}/store.db", base_path);
+        let store = Store::persistent(&store_path).expect("Failed to create persistent store");
+        DWebBackend {
+            path: base_path.to_string(),
+            port,
+            store,
+            veilid_api: None,
+        }
+    }
+
+    // Updated start method to initialize both Store and Veilid
+    pub async fn start(&mut self) -> eyre::Result<()> {
+        println!("Starting on {} with port {}", self.path, self.port);
+
+        // Ensure base directory exists
+        let base_dir = &self.path;
+        fs::create_dir_all(base_dir).await.map_err(|e| {
+            eyre::eyre!("Failed to create base directory {}: {}", base_dir, e)
+        })?;
+
+        // Initialize Veilid
+        let update_callback: Arc<dyn Fn(VeilidUpdate) + Send + Sync> = Arc::new(|update| {
+            info!("Received update: {:?}", update);
+        });
+
+        let mut rng = StdRng::from_entropy();
+        let random_suffix: u16 = rng.gen_range(10000..60000);
+
+        // Create a VeilidConfigInner instance
+        let config_inner = VeilidConfigInner {
+            program_name: format!("node{}", random_suffix),
+            namespace: format!("default_{}", random_suffix),
```

Comment:
What is the namespace used for? Is it combined with the program name? If not we should namespace to something like `openarchive` or some such

Reply:
Changed `namespace `to 'openarchive' and `program_name` to save-dweb-backend
I used the random_suffix in past tests and we might want to remove this now as it does no longer seem relevant.